src/gpu/nvidia/cudnn_pooling.hpp

```diff
@@ -41,7 +41,17 @@ struct cudnn_pooling_common_t {
 
         const auto src_size = src_wrap.nelems();
         const auto dst_size = dst_wrap.nelems();
-        const dims_t ws_size = {(dim_t)(src_size + dst_size)};
+
+        // Add a byte of extra memory in abx case to ensure the hash generated
+        // is different from the axb case. This is a workaround for issue #1727
+        const auto tag = get_tag(is_fwd ? *pd->dst_md() : *pd->diff_dst_md());
+        dim_t ws_offset = 0;
+        if (utils::one_of(tag, format_tag::ab, format_tag::abc,
+                    format_tag::abcd, format_tag::abcde)) {
+            ws_offset = 1;
+        }
+
+        const dims_t ws_size = {(dim_t)(src_size + dst_size + ws_offset)};
```

Comment:
The change doesn't solve the issue. The reproducer provided still shows an error.

Reply:
I looked at the issue and the fix one more time, and it seems this fix doesn't "solve" the issue properly - the knowledge about cache specifics gets into implementation.
I found two actual sources of the observed behavior:
1) cuDNN pooling impl [overwrites](https://github.com/oneapi-src/oneDNN/blob/master/src/gpu/nvidia/cudnn_pooling.hpp#L178) diff_src tensor despite the the fact it was already [initialized](https://github.com/oneapi-src/oneDNN/blob/master/src/common/pooling_pd.hpp#L320) as needed in [`set_default_formats()`](https://github.com/oneapi-src/oneDNN/blob/master/src/gpu/nvidia/cudnn_pooling.hpp#L156) call. And it does it wrong, not following the format.
2) Cache hit is still there if (1) gets fixed because just dst is not enough to distinguish between two cases, so src is needed there.

I open an internal PR to address this correctness problem with fixing both abovementioned issues.